src/main/java/se/liu/ida/hefquin/engine/datastructures/impl/SolutionMappingsHashTable.java

```diff
@@ -1,85 +1,171 @@
 package se.liu.ida.hefquin.engine.datastructures.impl;
 
-import java.util.Iterator;
+import java.util.*;
 
 import org.apache.jena.graph.Node;
 import org.apache.jena.sparql.core.Var;
 
+import org.apache.jena.sparql.engine.binding.Binding;
 import se.liu.ida.hefquin.engine.data.SolutionMapping;
+import se.liu.ida.hefquin.engine.data.impl.SolutionMappingUtils;
 
 public class SolutionMappingsHashTable extends SolutionMappingsIndexBase
 {
+	// Having List<Node> as key type for the hash table is probably
+	// not the best choice in terms of efficiency. However, it will have
+	// to do for the moment.
+	// TODO: can this be made more efficient?
+	protected final Map<List<Node>, List<SolutionMapping>> map = new HashMap<>();
+	protected final Set<Var> joinVariables;
+
+	public SolutionMappingsHashTable(final Set<Var> joinVariables){
+		this.joinVariables = joinVariables;
+	}
+
 	@Override
 	public int size() {
-		// TODO Auto-generated method stub
-		return 0;
+		return map.size();
 	}
 
 	@Override
 	public boolean isEmpty() {
-		// TODO Auto-generated method stub
-		return false;
+		return map.isEmpty();
```

Comment:
This may also be incorrect. Depending on how the `remove` method is implementation (e.g., in possible subclasses), there may be entries in `map` but they all have empty lists as their values. Hence, if `map` is not empty, you would still have to check whether there are non-empty buckets in the hash table.